internal/featuretests/httpproxy.go

```diff
@@ -16,11 +16,45 @@ package featuretests
 // HTTPProxy helpers
 
 import (
+	"strings"
+
 	projcontour "github.com/projectcontour/contour/apis/projectcontour/v1"
 )
 
-func prefixCondition(prefix string) []projcontour.Condition {
+// Returns an array of Conditions
+func prefixConditions(prefix string) []projcontour.Condition {
 	return []projcontour.Condition{{
 		Prefix: prefix,
 	}}
 }
+
+// Returns a single Condition
+func prefixCondition(prefix string) projcontour.Condition {
+	return projcontour.Condition{
+		Prefix: prefix,
+	}
+}
+
+func conditions(c ...projcontour.Condition) []projcontour.Condition { return c }
+
+func headerCondition(name, value, headerType string, present bool) projcontour.Condition {
```

Comment:
this helper doesn't seem to help much, you have to pass four parameters into it, some of which will be ignored depending on others. What about something like

```
headerContains(key, value string) projcontour.Condition
headerExists(key string) projcontour.Conditoin
```
and so forth

Reply:
sure thing